packages/input-option/src/InputOption/InputOption.style.ts

```diff
@@ -91,60 +69,176 @@ export const inputOptionWedge = css`
+      }
+
+      &,
+      & .${descriptionClassName} {
+        color: ${formThemeStyles[theme][state].description};
+      }
+
+      &,
+      & .${leftGlyphClassName} {
+        color: ${formThemeStyles[theme][state].leftGlyph};
+      }
+    `,
+    {
+      [wedgeStyles]: shouldShowWedge,
+      [hoverStyles]: !disabled && isInteractive,
+    },
+  );
+};
 
-    &:hover {
-      background-color: inherit;
-    }
+export const getMenuElementStyle = ({
+  theme,
+  checked,
+  highlighted,
+  disabled,
+  actionType,
+  showWedge,
+  isInteractive,
+}: {
+  theme: Theme;
+  checked?: boolean;
+  highlighted?: boolean;
+  disabled?: boolean;
+  actionType: ActionType;
+  showWedge?: boolean;
+  isInteractive?: boolean;
+}) => {
+  let state: State = 'default';
+
+  if (disabled) {
+    state = State.Disabled;
+  } else if (highlighted) {
+    state = State.Focus;
+  } else if (checked) {
+    state = State.Checked;
+  } else if (actionType === ActionType.Destructive) {
+    state = State.Destructive;
+  }
 
+  const wedge = css`
     &:before {
-      content: unset;
+      transform: scaleY(1) translateY(-50%);
+      background-color: ${menuThemeStyles[theme][state].wedge};
     }
-  `,
-  [Theme.Dark]: css`
-    cursor: not-allowed;
+  `;
 
-    &,
-    & .${descriptionClassName} {
-      color: ${palette.gray.dark1};
-    }
+  const shouldOverride = state === State.Checked || state === State.Destructive;
 
-    &:hover {
-      background-color: inherit;
-    }
+  const overrideHoverBgColor =
+    theme === Theme.Light ? palette.gray.dark3 : palette.gray.dark2;
 
-    &:before {
-      content: unset;
+  const hover = css`
+    ${hoverSelector} {
+      outline: none;
+      background-color: ${shouldOverride
+        ? overrideHoverBgColor
+        : menuThemeStyles[theme].hover.backgroundColor};
+
+      &,
+      & .${leftGlyphClassName} {
+        color: ${menuThemeStyles[theme].hover.leftGlyph};
+      }
+    }
+  `;
+
+  const titleHoverOverride = css`
+    ${hoverSelector} {
+      &,
+      & .${titleClassName} {
+        color: ${state === State.Destructive
+          ? palette.red.light2
+          : palette.white};
+      }
     }
-  `,
+  `;
+
+  return cx(
+    css`
+      background-color: ${menuThemeStyles[theme][state].backgroundColor};
+
+      .${titleClassName} {
+        color: ${menuThemeStyles[theme][state].title};
+        font-weight: ${checked && !disabled ? 'bold' : 'normal'};
+      }
+
+      &,
+      & .${descriptionClassName} {
+        color: ${menuThemeStyles[theme][state].description};
+      }
+
+      &,
+      & .${leftGlyphClassName} {
+        color: ${menuThemeStyles[theme][state].leftGlyph};
+      }
+    `,
+    {
+      [wedge]: showWedge && (state === 'focus' || state === 'checked'),
+      [hover]: !disabled && isInteractive,
+      [titleHoverOverride]:
+        shouldOverride && !disabled && theme === Theme.Dark && isInteractive,
+    },
+  );
 };
```

Comment:
Curious to hear others' thoughts on this, but there's a lot of repeated code here that feels very verbose in a function format. It looks like the `css` string is also the same across multiple functions, so that isn't necessary. I'd personally prefer a style object approach to writing this CSS (the way we handle mode styles in most components)

Reply:
there's too many conditions to make that work - we have theme.ts but to decide how to apply all of the stlyes, i think there's too much logic.

Reply:
I'm fine with the functional approach if there's a lot of branching. I would like to see these functions combined into one though, since I see a fair bit of overlap

Reply:
there's too much nuance and little rules for each to combine them into one - i can have a wrapper around both functions that just calls its respective if you'd like